content/en/docs/develop/porting.md

```diff
@@ -614,23 +614,333 @@ To make a patch:
    For example, if you have made one (`1`) patch only, export it like so:
 
    ```console
-   git format-patch HEAD~1
+   $ git format-patch HEAD~1
    ```
 
    This will save a new `.patch` file in the current directory; which should be the origin source files of `iperf3`.
 
 1. The next step is to create a `patches/` folder within the Unikraft port of the library and to move the new `.patch` file into this folder:
 
    ```console
-   mkdir ~/workspace/libs/iperf3/patches
-   mv ~/workspace/apps/iperf3/build/libiperf3/origin/iperf-3.10.1/*.patch ~/workspace/libs/iperf3/patches
+   $ mkdir ~/workspace/libs/iperf3/patches
+   $ mv ~/workspace/apps/iperf3/build/libiperf3/origin/iperf-3.10.1/*.patch ~/workspace/libs/iperf3/patches
    ```
 
-1. To register patches against Unikraft's build tool such that they are applied before the compilation of all source files, simply indicate it in the library's `Makefile.uk`:
+1. To register patches against Unikraft's build system such that they are applied before the compilation of all source files, simply indicate it in the library's `Makefile.uk`:
 
    ```Makefile
    # Add or edit ~/workspace/libs/iperf3/Makefile.uk
    LIBIPERF3_PATCHDIR = $(LIBIPERF3_BASE)/patches
    ```
 
 This concludes the necessary steps to port an application to Unikraft "from first principles".
+
+### Files Relevant to the Build System
```

Comment:
I'm not that happy that some information is presented above and then again here (`Makefile`, `Makefile.uk`, `Config.uk`).

Maybe we can just rename this section into `Internals of the Unikraft Build System` and then state that: "As we saw above, Unikraft uses a set of files as part of the configuration and build steps.
When porting an application, you want to be aware of these files and tune them to your needs.
This section delves into the internals of the Unikraft build system and how they are useful for porting an application."